pkg/cmd/server/api/types.go

```diff
@@ -0,0 +1,160 @@
+package api
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+// NodeConfig is the fully specified config starting an OpenShift node
+type NodeConfig struct {
+	api.TypeMeta
+
+	// NodeName is the value used to identify this particular node in the cluster.  If possible, this should be your fully qualified hostname.
+	// If you're describing a set of static nodes to the master, this value must match one of the values in the list
+	NodeName string
+
+	// ServingInfo describes how to start serving
+	ServingInfo ServingInfo
+
+	// MasterKubeConfig is a filename for the .kubeconfig file that describes how to connect this node to the master
+	MasterKubeConfig string
+
+	DNSDomain string
+	// ip:port
+	DNSAddress string
```

Comment:
only the IP portion of this is used (see `ClusterDNS:    net.ParseIP(options.DNSAddress),`) and if a host:port is set, it is not used at all (ParseIP returns nil)